utils/ixdtf/src/parsers/records.rs

```diff
@@ -211,6 +214,46 @@ pub enum TimeDurationRecord {
         /// Seconds value.
         seconds: u64,
         /// The parsed fraction value in nanoseconds.
-        fraction: u32,
+        fraction: Fraction,
     },
 }
+
+/// A fraction value in nanoseconds or lower value.
+///
+/// # Precision note
+///
+/// `ixdtf` parses a fraction value to a precision of 18 digits of precision, but
+/// preserves the fraction's digit length
+#[derive(Debug, Default, Clone, Copy, PartialEq)]
+#[allow(clippy::exhaustive_structs)] // A fraction is only a value and its digit length.
+pub struct Fraction {
+    // The count of fraction digits, i.e. the fraction's digit length.
+    pub digits: u8,
+    // The parsed fraction value.
+    pub value: u64,
+}
+
+impl Fraction {
+    /// Returns Some(`u32`) representing the `Fraction` as it's computed
+    /// nanosecond value or `None` if the digits exceeds 9 digits.
+    pub fn to_nanoseconds(&self) -> Option<u32> {
+        if self.digits <= 9 {
+            10u32
+                .checked_pow(9 - u32::from(self.digits))
+                .map(|x| x * self.value as u32)
+        } else {
+            None
+        }
+    }
+
+    /// Returns a `u64` representing the `Fraction` as it's computed
+    /// nanosecond value, truncating any value beyond 9 digits to
+    /// nanoseconds
+    pub fn to_truncated_nanoseconds(&self) -> u64 {
+        if self.digits <= 9 {
+            self.value
+        } else {
+            self.value / 10u64.pow(u32::from(self.digits - 9))
+        }
```

Comment:
Issue: Something here doesn't look right. The code for the `self.digits <= 9` case should be the same in both functions I think. Please make sure this is tested.

Reply:
Huh, that's true. I think when I initially wrote that, it was with the idea of truncating on all cases, but that doesn't align with the function name.

I've updated the behavior and added some doc tests and tests for potentially malformed `Fraction`s since the fields are public.